### ISSUE TITLE
unhandled exceptin during `ng update`

### ISSUE THREAD

**Issue by j-tesla** [angular/angular-cli#20963]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [x] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

 Yes, the previous version in which this bug was not present was: ....

### Description

 A clear and concise description of the problem...
`ng update` fails with an error.
## 🔬 Minimal Reproduction
```sh
ng update
```
Installed packages: (`npm list`)
```

├── @angular-devkit/build-angular@0.1102.13
├── @angular/animations@11.2.14
├── @angular/cdk@11.2.13
├── @angular/cli@11.2.13
├── @angular/common@11.2.14
├── @angular/compiler-cli@11.2.14
├── @angular/compiler@11.2.14
├── @angular/core@11.2.14
├── @angular/forms@11.2.14
├── @angular/material@11.2.13
├── @angular/platform-browser-dynamic@11.2.14
├── @angular/platform-browser@11.2.14
├── @angular/router@11.2.14
├── @types/jasmine@3.6.11
├── @types/node@12.20.13
├── codelyzer@6.0.2
├── jasmine-core@3.6.0
├── jasmine-spec-reporter@5.0.2
├── karma-chrome-launcher@3.1.0
├── karma-coverage@2.0.3
├── karma-jasmine-html-reporter@1.6.0
├── karma-jasmine@4.0.1
├── karma@6.1.2
├── protractor@7.0.0
├── rxjs@6.6.7
├── ts-node@8.3.0
├── tslib@2.2.0
├── tslint@6.1.3
├── typescript@4.1.5
└── zone.js@0.11.4
```

## 🔥 Exception or Error

![image](https://user-images.githubusercontent.com/58279279/119797732-206b4580-bef8-11eb-94fb-58ccd9245e4e.png)

<pre><code>

</code></pre>

## 🌍 Your Environment

<pre><code>

Your global Angular CLI version (12.0.0) is greater than your local version (11.2.13). The local Angular CLI version is used.

To disable this warning use "ng config -g cli.warnings.versionMismatch false".

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 11.2.13
Node: 14.17.0
OS: linux x64

Angular: 11.2.14
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.13
@angular-devkit/build-angular   0.1102.13
@angular-devkit/core            11.2.13
@angular-devkit/schematics      11.2.13
@angular/cdk                    11.2.13
@angular/cli                    11.2.13
@angular/material               11.2.13
@schematics/angular             11.2.13
@schematics/update              0.1102.13
rxjs                            6.6.7
typescript                      4.1.5

</code></pre>

**Anything else relevant?**
OS: Pop!_OS 20.04 LTS x86_64

**Comment by mlbiche** [comment]
I had the same. It was because my global installation of `@angular/cli` was not on the same version (already bumped to v12 while my local `@angular/cli` was in version 11).
Maybe make sure you have `@angular/cli` installed globally and that your local and global installations are on the same version ?

**Comment by alan-agius4** [comment]
Please update your global CLI to the latest version.

```
npm i -g @angular/cli
```

**Comment by clydin** [comment]
The issue was caused by a defect in 12.0.0 which was fixed in 12.0.1.  Installing the latest version of the CLI globally first will solve the issue.
The local and global versions do not need to be the same.  Installing the latest global first will actually remove the need for the local version to download and install a newer version during the update initialization process.

**Comment by Splaktar** [comment]
Oh thank goodness there is a fix for this. It's been driving me nuts trying to debug this for the last hour+.
